packages/x-data-grid/src/hooks/features/pagination/useGridPaginationModel.ts

```diff
@@ -67,11 +73,15 @@ export const useGridPaginationModel = (
     | 'pagination'
     | 'signature'
     | 'rowHeight'
+    | 'resetPageOnSortFilter'
   >,
 ) => {
   const logger = useGridLogger(apiRef, 'useGridPaginationModel');
 
+  const filterModel = useGridSelector(apiRef, gridFilterModelSelector);
```

Comment:
Instead of the reactive binding, would it be better to directly call the selection for ref initialization?
In `handleFilterModelChange` we could directly receive the `filterModel` parameter to avoid recomputing the method.